hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1443,6 +1446,10 @@ func (r *NodePoolReconciler) reconcileMachineDeployment(log logr.Logger,
 	// is at the expected version (spec.version) and config (userData Secret) so we reconcile status and annotation.
 	if MachineDeploymentComplete(machineDeployment) {
 		if nodePool.Status.Version != targetVersion {
+			if nodePool.Status.Version == "" {
+				metrics.RecordNodePoolInitialRolloutDuration(nodePool)
+			}
+
```

Comment:
You'd also need to do the same in https://github.com/openshift/hypershift/blob/b0ab52df6473620696f5e7e4b26b7fefea1db832/hypershift-operator/controllers/nodepool/inplace.go#L150-L151 
for inplace nodepools